packages/blitz/src/bin/cli.ts

```diff
@@ -2,72 +2,102 @@ import * as path from "path"
+  let blitzPkgPath
+  if (isInDevelopmentAsGloballyLinked) {
+    blitzPkgPath = globalBlitzPath
+  } else {
+    // localBlitzPath won't exist if used outside a blitz app directory
+    blitzPkgPath = localBlitzPath || globalBlitzPath
+  }
 
-const cli = require(cliPkgPath)
+  const cliPkgPath = resolveFrom(blitzPkgPath, "@blitzjs/cli")
 
-const options = require("minimist")(process.argv.slice(2))
-const hasVersionFlag = options._.length === 0 && (options.v || options.version)
-const hasVerboseFlag = options._.length === 0 && (options.V || options.verbose)
+  const cli = require(cliPkgPath)
 
-if (hasVersionFlag) {
-  if (hasVerboseFlag) {
-    console.log("debug: blitzPkgPath:", blitzPkgPath)
-    console.log("debug: cliPkgPath:", cliPkgPath, "\n")
-  }
-  try {
-    const osName = require("os-name")
-    console.log(`${osName()} | ${process.platform}-${process.arch} | Node: ${process.version}\n`)
+  const options = require("minimist")(process.argv.slice(2))
+  const hasVersionFlag = options._.length === 0 && (options.v || options.version)
+  const hasVerboseFlag = options._.length === 0 && (options.V || options.verbose)
 
-    const globalBlitzPkgJsonPath = pkgDir.sync(globalBlitzPath) as string
-    const localBlitzPkgJsonPath = pkgDir.sync(localBlitzPath)
-
-    if (globalBlitzPkgJsonPath !== localBlitzPkgJsonPath) {
-      // This branch won't run if user does `npx blitz` or `yarn blitz`
-      const globalVersion = require(path.join(globalBlitzPkgJsonPath, "package.json")).version
-      console.log(`blitz: ${globalVersion} (global)`)
+  if (hasVersionFlag) {
+    if (hasVerboseFlag) {
+      console.log("debug: blitzPkgPath:", blitzPkgPath)
+      console.log("debug: cliPkgPath:", cliPkgPath, "\n")
     }
+    try {
+      const osName = require("os-name")
+      console.log(`${osName()} | ${process.platform}-${process.arch} | Node: ${process.version}\n`)
 
-    if (localBlitzPkgJsonPath) {
-      const localVersion = require(path.join(localBlitzPkgJsonPath, "package.json")).version
-      console.log(`blitz: ${localVersion} (local)`)
-    }
+      const globalBlitzPkgJsonPath = pkgDir.sync(globalBlitzPath) as string
+      const localBlitzPkgJsonPath = pkgDir.sync(localBlitzPath)
+
+      if (globalBlitzPkgJsonPath !== localBlitzPkgJsonPath) {
+        // This branch won't run if user does `npx blitz` or `yarn blitz`
+        const globalVersion = require(path.join(globalBlitzPkgJsonPath, "package.json")).version
+        console.log(`blitz: ${globalVersion} (global)`)
+      }
+
+      if (localBlitzPkgJsonPath) {
+        const localVersion = require(path.join(localBlitzPkgJsonPath, "package.json")).version
+        console.log(`blitz: ${localVersion} (local)`)
+      }
 
-    console.log("") // One last new line
-  } catch (e) {
-    console.log("blitz error", e)
+      await printEnvInfo()
+
+      console.log("") // One last new line
+    } catch (e) {
+      throw new Error(`Blitz Error: ${e}`)
+    }
+    process.exit(0)
```

Comment:
Since the call to `envinfo` is async, this line is causing the script to exit before it can finish, leading to no output being printed. 

An alternative would be to move process.exit into the callback of envinfo? Though if other async stuff needs to be added down the road we might run into the same problem again.